kafka-connect-s3/src/test/java/io/confluent/connect/s3/storage/S3StorageTest.java

```diff
@@ -0,0 +1,172 @@
+    AmazonServiceException e = new AmazonServiceException("Retriable exception");
+    e.setStatusCode(HttpStatus.SC_INTERNAL_SERVER_ERROR);
+    assertTrue(retryPolicy.getRetryCondition().shouldRetry(null, e, 1));
+  }
+
+  @Test
+  public void testRetryPolicyNonRetriableServiceException() throws Exception {
+    AmazonServiceException e = new AmazonServiceException("Non-retriable exception");
+    e.setStatusCode(HttpStatus.SC_METHOD_NOT_ALLOWED);
+    assertFalse(retryPolicy.getRetryCondition().shouldRetry(null, e, 1));
+  }
+
+  @Test
+  public void testRetryPolicyRetriableThrottlingException() throws Exception {
+    AmazonServiceException e = new AmazonServiceException("Retriable exception");
+    e.setErrorCode("TooManyRequestsException");
+    assertTrue(retryPolicy.getRetryCondition().shouldRetry(null, e, 1));
+  }
+
+  @Test
+  public void testRetryPolicyRetriableSkewException() throws Exception {
+    AmazonServiceException e = new AmazonServiceException("Retriable exception");
+    e.setErrorCode("RequestExpired");
+    assertTrue(retryPolicy.getRetryCondition().shouldRetry(null, e, 1));
+  }
+
+  @Test
+  public void testRetryPolicyDelayRanges() throws Exception {
+    assertComputeRetryInRange(10, 10L);
+    assertComputeRetryInRange(10, 100L);
+    assertComputeRetryInRange(10, 1000L);
+    assertComputeRetryInRange(MAX_RETRIES + 1, 1000L);
+    assertComputeRetryInRange(100, S3_RETRY_MAX_BACKOFF_TIME_MS + 1);
+    assertComputeRetryInRange(MAX_RETRIES + 1, S3_RETRY_MAX_BACKOFF_TIME_MS + 1);
+  }
+
+  /**
+   * Calculates exponential delay, capped by
+   * {@link com.amazonaws.retry.PredefinedBackoffStrategies#MAX_RETRIES} number of retries
+   * and {@link io.confluent.connect.s3.S3SinkConnectorConfig#S3_RETRY_MAX_BACKOFF_TIME_MS} total delay time
+   * in ms
+   *
+   * @param retriesAttempted
+   * @param baseDelay
+   * @return
+   * @see PredefinedBackoffStrategies#calculateExponentialDelay(int, int, int)
+   */
+  private int calculateExponentialDelay(
+      int retriesAttempted, long baseDelay
+  ) {
+    int retries = Math.min(retriesAttempted, MAX_RETRIES);
+    return (int) Math.min((1L << retries) * baseDelay, S3_RETRY_MAX_BACKOFF_TIME_MS);
+  }
+
+  private void assertComputeRetryInRange(
+      int retryAttempts,
+      long retryBackoffMs
+  ) throws Exception {
+
+    localProps.put(S3_RETRY_BACKOFF_CONFIG, String.valueOf(retryBackoffMs));
+    setUp();
+    RetryPolicy.BackoffStrategy backoffStrategy = retryPolicy.getBackoffStrategy();
+
+    for (int i = 0; i != 20; ++i) {
+      for (int retries = 0; retries <= retryAttempts; ++retries) {
+        long maxResult = calculateExponentialDelay(retries, retryBackoffMs);
+        long result = backoffStrategy.delayBeforeNextRetry(null, null, retries);
+        if (retryBackoffMs < 0) {
+          assertEquals(0, result);
+        } else {
+          assertTrue(result >= 0L);
+          assertTrue(result <= maxResult);
+        }
+      }
+    }
+  }
+}
```

Comment:
minor: no newline at the end of the file.

Reply:
oops, fixed.